model/flow/version_beacon.go

```diff
@@ -155,3 +156,39 @@ func (v *VersionBeacon) String() string {
 	}
 	return buffer.String()
 }
+
+// ProtocolStateVersionUpgrade is a service event emitted by the FlowServiceAccount
+// to signal an upgrade to the Protocol State version. `NewProtocolStateVersion`
+// must be strictly greater than the currently active Protocol State Version,
+// otherwise the service event is ignored.
+// If the node software supports `NewProtocolStateVersion`, then it begins using
+// this Protocol State Version, beginning with the first block B where BOTH:
+//  1. The `ProtocolStateVersionUpgrade` service event has been sealed in B's ancestry
+//  2. B.view >= `ActiveView`
+//
+// NOTE: The above implies that, if the `ProtocolStateVersionUpgrade` event is sealed in a
+// block with view >= `ActiveView`, then `NewProtocolStateVersion` takes effect immediately.
+//
+// Otherwise, the node software stops processing blocks, until it is manually updated
```

Comment:
I don't think we should immediately make an issue, but it's worth considering. I will add it as a discussion point to the FLIP/discussion post (at the moment I can't edit the post...). 

The reason I don't think we should necessarily pre-emptively do this:
- Adding a flag that disable startup version checking would be a relatively small change, I think. Not having this in place preemptively would not be a big impediment to recovery.
- If we do need this, my guess is that it would be necessary but not sufficient. We will need to evaluate how the different versions will interact, and possibly make some changes to the software in one or the other version, before recovering.

Reply:
thanks for the explanation. Agree with all of your points.